airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/java/io/airbyte/cdk/testutils/ContainerFactory.java

```diff
@@ -102,39 +119,79 @@ public final C shared(String imageName, String... methods) {
    * Returns an exclusive instance of the testcontainer.
    */
   @SuppressWarnings("unchecked")
+  @Deprecated
   public final C exclusive(String imageName, String... methods) {
-    return (C) createAndStartContainer(DockerImageName.parse(imageName), Stream.of(methods).toList());
+    return exclusive(imageName,
+        (NamedContainerModifier<C>) Stream.of(methods).map(n -> new NamedContainerModifierImpl<C>(n, resolveModifierByName(n))).toList());
+  }
+
+  public final C exclusive(String imageName) {
+    return exclusive(imageName, new ArrayList<>());
+  }
+
+  public final C exclusive(String imageName, NamedContainerModifier<C>... namedContainerModifiers) {
+    return exclusive(imageName, List.of(namedContainerModifiers));
+  }
+
+  public final C exclusive(String imageName, List<NamedContainerModifier<C>> namedContainerModifiers) {
+    return (C) createAndStartContainer(DockerImageName.parse(imageName), namedContainerModifiers);
+  }
+
+  public interface NamedContainerModifier<C extends GenericContainer<?>> {
+
+    String name();
+
+    Consumer<C> modifier();
+
   }
 
-  private GenericContainer<?> createAndStartContainer(DockerImageName imageName, List<String> methodNames) {
-    LOGGER.info("Creating new shared container based on {} with {}.", imageName, methodNames);
-    try {
-      GenericContainer<?> container = createNewContainer(imageName);
-      final var methods = new ArrayList<Method>();
-      for (String methodName : methodNames) {
-        methods.add(getClass().getMethod(methodName, container.getClass()));
+  public record NamedContainerModifierImpl<C extends GenericContainer<?>> (String name, Consumer<C> method) implements NamedContainerModifier<C> {
+
+    public String name() {
```

Comment:
Can you configure your editor to add `@Override` annotations please?

Reply:
It's already enabled. I don't think it applies on interface overrides